lib/workers/repository/package-files.ts

```diff
@@ -24,7 +26,72 @@ export class PackageFiles {
     this.data.clear();
   }
 
-  public static getDashboardMarkdown(config: RenovateConfig): string {
+  /**
+   * Truncates the detected dependencies' section until it fits the available space
+   * i.e. It has length smaller than maxLength.
+   * This does not mutate the original PackageFiles data
+   * Note:  setHeader=false is used for testing purposes only
+   *        Mainly for comparing the output of getTruncatedMarkdown and getDashboardMarkdown
+   * @param config
+   * @param maxLength
+   * @param setHeader
+   */
+  public static getTruncatedMarkdown(
+    config: RenovateConfig,
+    maxLength: number,
+    setHeader = true
+  ): string {
+    const note =
+      '> **Note**\n> Detected dependencies section has been truncated\n';
+    const title = `## Detected dependencies\n\n`;
+
+    // exclude header length from the available space
+    const maxHeaderLen = setHeader ? (title + note).length : 0;
```

Comment:
Is there a case where `maxLength - title` is long enough, but `maxLength - title - note` is too small? I understand you're pessimistically setting absolute max length.

Reply:
Seems like an edge case, will happen when 
`maxLength  - note < len + title < maxLength`.
where `note.len` is a `const` and is =~30. 
I don't think we should handle it for the cost of the added complexity